src/main/java/com/sarality/db/query/QueryParser.java

```diff
@@ -0,0 +1,56 @@
+package com.sarality.db.query;
+
+import android.text.TextUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Given a query, reverses the Query Builder operations and extracts the columnlist, operators and arguments
+ *
+ * @author satya@ Satya Puniani
+ */
+public class QueryParser {
+
+  //TODO strongly type these lists to Column and Operator
+  private final LogicalOperator operator = LogicalOperator.AND;
+  private final List<String> columnList = new ArrayList<>();
+  private final List<String> operatorList = new ArrayList<>();
+  private final List<String> argumentValueList = new ArrayList<>();
+
+  public QueryParser(Query query){
+
+    //TODO get logical operator
+
+    //get where clause as string array
+    String operatorNames = TextUtils.join("|",LogicalOperator.values());
+    String[] whereClause = query.getWhereClause().split(operatorNames);
+
+    //for each where clause, get columnname, operator, argument
```

Comment:
Please add a TODO here to refine this further and throw ParseException when the class is not able to process the query